sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService, ESDCIntegrationConfig } from "@sims/utilities/config";
+import { StudentLoanBalancesIntegrationService } from "./student-loan-balances.integration.service";
+import { StudentLoanBalancesSFTPResponseFile } from "./models/student-loan-balances.model";
+import { StudentService } from "@sims/integrations/services";
+import * as path from "path";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import { getISODateOnlyString } from "@sims/utilities";
+import { DataSource } from "typeorm";
+import { StudentLoanBalances } from "@sims/sims-db";
+
+/**
+ * Manages to process the Student Loan Balances files
+ * once they are available in the SFTP location.
+ */
+@Injectable()
+export class StudentLoanBalancesProcessingService {
+  private readonly esdcConfig: ESDCIntegrationConfig;
+  constructor(
+    private readonly dataSource: DataSource,
+    config: ConfigService,
+    private readonly studentLoanBalancesIntegrationService: StudentLoanBalancesIntegrationService,
+    private readonly studentService: StudentService,
+  ) {
+    this.esdcConfig = config.esdcIntegration;
+  }
+
+  /**
+   * Download all files from SFTP and process them all.
+   * @param parentProcessSummary parent process summary.
+   * @returns Summary with what was processed and the list of all errors, if any.
+   */
+  async processStudentLoanBalances(
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    // Process summary to be populated by each enqueueing workflow call.
+    const serviceProcessSummary = new ProcessSummary();
+    const remoteFilePaths =
+      await this.studentLoanBalancesIntegrationService.getResponseFilesFullPath(
+        this.esdcConfig.ftpResponseFolder,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.PT.MBAL.[0-9]{8}.*`,
+          "i",
+        ),
+      );
+    for (const remoteFilePath of remoteFilePaths) {
+      await this.processFile(remoteFilePath, serviceProcessSummary);
+      parentProcessSummary.children(serviceProcessSummary);
```

Comment:
Every for loop is supposed to create its own `ProcessSummary` and then this newly created `ProcessSummary` will be added as a `children`.
Is the log right now produced as expected?